velox/type/Type.cpp

```diff
@@ -443,13 +443,19 @@ bool RowType::equivalent(const Type& other) const {
 }
 
 bool RowType::equals(const Type& other) const {
-  if (!this->equivalent(other)) {
+  if (&other != this && !Type::hasSameTypeId(other)) {
```

Comment:
Let's write it like RowType::equivalent

```
if (&other == this) {
    return true;
  }
  if (!Type::hasSameTypeId(other)) {
    return false;
  }
```

Reply:
Noob question: `&other == this` here calls the reloaded operator or compares the address?

Reply:
This compares raw pointers. Doesn't call operator== on the object.